src/wallet/wallet.cpp

```diff
@@ -1544,7 +1644,7 @@ bool CWalletTx::RelayWalletTransaction(CConnman* connman)
     assert(pwallet->GetBroadcastTransactions());
     if (!IsCoinBase())
     {
-        if (GetDepthInMainChain() == 0 && !isAbandoned() && InMempool()) {
+        if (GetDepthInMainChain() == 0 && !isAbandoned() && (InMempool() || fSPV)) {
```

Comment:
I don't understand in general what the resend wallet transaction code is for, and probably this isn't the place to explain it, but it might be nice to have a comment here saying how the SPV condition relates.